src/code/main.py

```diff
@@ -107,7 +134,46 @@ def _delete_snap(configs_iter: Iterable[configs.Config], path_suffix: str, sync:
         os_utils.eprint(f"Target {path_suffix} not found in any config.")
 
 
-def _config_operation(command: str, source: str, comment: str, sync: bool):
+def _batch_delete_snaps(
+    configs_iter: Iterable[configs.Config],
+    scope: tuple[str, str],
+    indicator: str,
+    sync: bool,
+):
+    configs_iter = list(configs_iter)
```

Comment:
nit: Change the argument to configs_iter: list[configs.config]?
Or alternatively, instead of re-assigning in place, create a new `configs_list = list(configs_iter)` here.
Reason - mostly for code legibility purposes, makes it clear that the new variable is a list. Theoretical drawback that python will keep the old config_iter is pretty negligible. You can choose to `del configs_iter` after so that it is not inadvertently used afterwards.

Reply:
Thank you for pointing that out. The change of the data type of `configs_iter` from `Iterable[configs.config]` to `list[configs.config]` is indeed confusing. No problem, I will change `configs_iter = list(configs_iter)` to `configs_list = list(configs_iter)`.